utils/fwutils.py

```diff
@@ -234,6 +259,139 @@ def get_images(self) -> tuple[str, str]:
+
+                self.logger.error("Couldn't find the board_config file, exitting...")
+                exit(1)
+        else:
+            self.logger.error(
+                f"server {self.imc_address} with number {number} doesn't have pre built fixboard images yet, please add the necessary files to the repo"
+            )
+            exit(1)
+
+    def apply_fixboard(self) -> None:
+        fixboard_bin_board_config_file = self.ensure_fixboard_image_on_imc()
+        full_address = f"root@{self.imc_address}"
+        result = ssh_run(
+            "flash_erase /dev/mtd0 0x30000 1",
+            full_address,
+            dry_run=self.dry_run,
+        )
+        if result.returncode:
+            self.logger.error(f"Couldn't flash_erase, Error: {result.err}")
+            exit(1)
+
+        result = ssh_run(
+            f"nandwrite --start=0x30000 --input-size=0x1000 -p /dev/mtd0 {fixboard_bin_board_config_file}",
+            full_address,
+            dry_run=self.dry_run,
+        )
+        if result.returncode:
+            self.logger.error(f"Couldn't nandwrite, Error: {result.err}")
+            exit(1)
+        self.logger.info("Rebooting IMC now!")
+        result = ssh_run(
+            "reboot",
+            full_address,
+            dry_run=self.dry_run,
+        )
+
+    def fixboard_is_needed(self) -> Optional[bool]:
+        full_address = f"root@{self.imc_address}"
+        try:
+            result = ssh_run(
+                "iset-cli get-board-config",
+                full_address,
+                dry_run=self.dry_run,
+            )
+            if result.returncode:
+                self.logger.error(
+                    f"Couldn't retrieve get board config using iset-cli, Error: {result.err}"
+                )
+                exit(1)
+            # Parse the JSON from the command output
+            self.logger.debug(f"Board Config command output: {result.out}")
+            board_config: dict[str, str] = json.loads(result.out)
+            self.logger.debug(f"Json Board Config: {board_config}")
+            for key, value in board_config.items():
+                if "MAC Address" in key:
+                    if value in ["00:00:00:00:00:00", "FF:FF:FF:FF:FF:FF"]:
+                        return True
+                if "PBA" in key:
+                    if value in [
+                        "000000000000000000000000",
+                        "FFFFFFFFFFFFFFFFFFFFFFFF",
+                    ]:
+                        self.logger.debug(f"PBA check failed for {key}: {value}")
+                        return True
+                if "Serial Number" in key:
+                    if value == "":
+                        return True
+
+            return False
+
+        except json.JSONDecodeError as e:
+            self.logger.error(f"Failed to parse JSON: {e}")
+            return None
+
+        except Exception as e:
+            self.logger.error(f"Unexpected error: {e}")
+            return None
```

Comment:
I think we can just bail here, rather than returning None and having to handle this function returning either a boolean or None.

This is messy. If we encounter an error that we don't know how to handle, the hardware is in an unexpected state, and it is ok to just log the error, and exit immediately.

Reply:
Updated with the changes and tested